src/citrine/informatics/workflows/design_workflow.py

```diff
@@ -62,3 +71,36 @@ def design_executions(self) -> DesignExecutionCollection:
             raise AttributeError('Cannot initialize execution without project reference!')
         return DesignExecutionCollection(
             project_id=self.project_id, session=self._session, workflow_id=self.uid)
+
+    @property
+    @deprecated(deprecated_in="2.42.0", removed_in="3.0.0",
+                details="Please use the branch_root_id and branch_version instead.")
+    def branch_id(self):
+        """[deprecated] Retrieve the version ID of the branch this workflow is on."""
+        return self._branch_id
+
+    @branch_id.setter
+    @deprecated(deprecated_in="2.42.0", removed_in="3.0.0",
+                details="Please set the branch_root_id and branch_version instead.")
+    def branch_id(self, value):
+        self._branch_id = value
```

Comment:
Hmmmmmmmm...I think this is an oversight on my part. They should still be able to set `branch_id` and have the exact same impact as before. So yes, setting it should blank `branch_root_id` and `branch_version`. And when registering or updating, we should use it if the root and version aren't set, and emit a deprecation warning.

Reply:
Ah, I see where I got crossed up. It respects it for `update`, but not for `register`.

Reply:
Ah, actually. I still think setting `branch_id` should mark the root ID and version as `None`. But note that it's actually handled properly in the collection. That is, when updating, it's used if `branch_root_id` and `branch_version` are `None`. And when registering, the version ID should come from the collection, not the payload.

The issue is that I mistakenly broke that element of the collection. I removed the `branch_id` field of the collection, instead of deprecating it. So I need to add it back, then handle it properly in the collection methods.